src/relay/op/tensor/transform.cc

```diff
@@ -1428,6 +1428,58 @@ RELAY_REGISTER_OP("meshgrid")
     .set_attr<FTVMCompute>("FTVMCompute", MeshgridCompute)
     .set_attr<TOpPattern>("TOpPattern", kInjective);
 
+// interpolate operator
+TVM_REGISTER_NODE_TYPE(InterpolateAttrs);
+
+bool InterpolateRel(const Array<Type>& types, int num_inputs, const Attrs& raw_attrs,
+                    const TypeReporter& reporter) {
+  // 'types' contains: [x, xp, fp, result]
+  CHECK_EQ(types.size(), 4);
+  const InterpolateAttrs* attrs = raw_attrs.as<InterpolateAttrs>();
+  CHECK(attrs != nullptr);
+  const auto* x = types[0].as<TensorTypeNode>();
+  if (x == nullptr) {
+    return false;
+  }
+  const auto* xp = types[1].as<TensorTypeNode>();
+  if (xp == nullptr) {
+    return false;
+  }
+  const auto* fp = types[2].as<TensorTypeNode>();
+  if (fp == nullptr) {
+    return false;
+  }
+
+  const auto& x_shape = x->shape;
+  const auto& xp_shape = xp->shape;
+  const auto& fp_shape = fp->shape;
+
+  for (size_t i = 0; i < xp_shape.size(); i++) {
+    CHECK(reporter->AssertEQ(xp_shape[i], fp_shape[i]))
+        << "xp and fp must have the same shape: " << xp_shape << " vs " << fp_shape;
+  }
+
+  reporter->Assign(types[3], TensorType(x_shape, x->dtype));
```

Comment:
Should the output type be `fp->dtype` instead? Since indices could be a different type from the data.

Reply:
That makes sense, thanks.